### PR TITLE
feat(protocol-designer): Add delay field validation

### LINKED ISSUE #6008: feat: PD Validation- Delay
# Overview
As a PD user I would like to create only valid Delays in my protocol

# Implementation Details

Time:
- Min: 1
- Max: None
- Masking: no special characters, no decimals, cannot start with “0”

Tip position:
- Same min, max, and masking as Aspirate/dispense level tip position

# Design

N/A- there is no visualized error state of the time field due to masking

### LINKED ISSUE #6018: feat: PD Sensible default- Delay
# Overview
As a PD user I would like my advanced settings to have sensible defaults

# Implementation Details

- Time: 1s
- Tip position: Default is whatever the aspirate/dispense tip position is

### LINKED ISSUE #6108: feat: PD Form fields- Delay tip positioning modals
## Overview

The delay `mm` field should not be a normal number field, but like the other tip position fields, should display a number that cannot be edited directly in that field. When clicked, the field should open a tip positioning modal that is tied to that field.

## Acceptance criteria

For both aspirate delay and dispense delay `mm` fields:

- [ ] Tip positioning modal should open, corresponding to the field (modal's value pre-populates with field value, saving the modal updates the field value)
- [ ] Should have min/max range specified in #6008

### PR DESCRIPTION
**Pull request by Kadee80**
# Overview

Wont merge this till we are done with 4.0.4 but... closes #6008 closes #6108 closes #6018 

Makes tip position for delay a `TipPositionField` (so modal shows and default is shown) adds a prop/util to hide the label 'tip position' and validates delay fields.

# Changelog

- feat(protocol-designer): Add delay field validation

# Review requests

- [ ] Validation matches ticket
- [ ] Modal opens and text is sane
- [ ] delay tip pos field populates with default

# Risk assessment

low. PD behind a FF

### CODE REVIEW

File: protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js [hunk 1]
```diff
@@ -15,7 +15,10 @@ import { getDisabledFields } from '../../../../steplist/formLevel'
 import stepFormStyles from '../../StepEditForm.css'
 import styles from './TipPositionInput.css'
 import { TipPositionModal } from './TipPositionModal'
-import { getIsTouchTipField } from '../../../../form-types'
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:18]: nitpick: not introduced by this PR but I think all imports from ` '../../../../form-types'` should go above `TipPositionModal` and the styles. Whatever https://github.com/Opentrons/opentrons/pull/4680#discussion_r363102698 says

File: protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js [hunk 2]
```diff
@@ -101,13 +107,14 @@ function TipPositionInput(props: Props) {
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:107]: nitpick: also not introduced by this PR but might be worth making this exact typed

**Review by shlokamin — COMMENTED** [review]
Left a few nitpick comments, but also I noticed that you can have `0`s prepend the time field, which https://github.com/Opentrons/opentrons/issues/6008 says it shouldn't. Idk if we've ever done this sort of masking before... but we might have. It sounds really annoying tbh. 

Screen Shot 2020-07-16 at 10 12 43 AM

File: protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js (hunk 2 above)
Comment (Kadee80) [reply protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:108]: didn't do cause of dispatch props

Comment (shlokamin) [reply protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:108]: what do you mean? isn't `WrapperProps` different for that reason?

**Comment by Kadee80** [comment]
> Left a few nitpick comments, but also I noticed that you can have `0`s prepend the time field, which #6008 says it shouldn't. Idk if we've ever done this sort of masking before... but we might have. It sounds really annoying tbh.
> 
> Screen Shot 2020-07-16 at 10 12 43 AM

Yeah we dont do this masking anywhere - see volume field masking/validation

File: protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js (hunk 2 above)
Comment (Kadee80) [reply protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:108]: sorry i may have misunderstood the comment based on the location

Comment (shlokamin) [reply protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:108]: oh sorry that's my bad... i'm always a few lines off when making these comments for some reason 🤦🏽‍♂️

**Comment by shlokamin** [comment]
I can't remember what we decided, but now that the field level error is gone you can enter `0` for the seconds field. Is this expected?

**Comment by Kadee80** [comment]
> I can't remember what we decided, but now that the field level error is gone you can enter `0` for the seconds field. Is this expected?

Now that I removed the Min field error, 0 is allowed. I wonder if we can get the defaultTo(1) to work if that would be acceptable?

**Review by shlokamin — APPROVED** [review]
LGTM

**Review by IanLondon — COMMENTED** [review]
Just want to keep track of changes we made to the associated tickets:

- #6008 says `Delay > seconds` min is 1, but we changed it to: no min (min is 0 because of input masking). Then there's no error validation for the field, just masking :heavy_check_mark: 
- #6018 seems achieved here, but do we have a ticket for "when you change the labware, the associated delay tip position field should reset"?

File: protocol-designer/src/steplist/formLevel/getDefaultsForStepType.js [hunk 3]
```diff
@@ -10,7 +11,12 @@ import {
 } from '../../constants'
 import type { StepType, StepFieldName } from '../../form-types'
 
+// OT_PD_ENABLE_AIR_GAP_AND_DELAY
```
Comment (IanLondon) [inline protocol-designer/src/steplist/formLevel/getDefaultsForStepType.js:14]: extra comment

**Review by IanLondon — APPROVED** [review]
left some comments to make sure we aren't missing tickets

**Pull request merged**